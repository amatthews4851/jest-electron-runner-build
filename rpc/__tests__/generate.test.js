/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import {prepareFiles} from '../src/generate';
import path from 'path';

test('generation logic', async () => {
  const result = await prepareFiles({
    globs: [path.resolve(__dirname, '../__fixtures__/generation/*.js')],
  });
  expect(result).toHaveLength(1);
  expect(result[0][1]).toMatchSnapshot();
});
