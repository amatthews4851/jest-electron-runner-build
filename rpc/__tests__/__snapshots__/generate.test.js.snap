// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generation logic 1`] = `
"/**
 * ****************************************************
 * THIS IS A GENERATED FILE. DO NOT MODIFY IT MANUALLY!
 * ****************************************************
 * @flow
 * @generated 67207e6e69ae37366b3c421b7eda63f8
 */

import typeof Methods from './SomeRPC.js';
import RPCProcess from '@jest-runner/rpc/RPCProcess';

class SomeRPCProcess extends RPCProcess<Methods> {
  initializeRemote(): Methods {
    return {
      test1: (this.jsonRPCCall.bind(this, 'test1'): any),
      test2: (this.jsonRPCCall.bind(this, 'test2'): any),
    };
  }
}
module.exports = SomeRPCProcess;
"
`;
